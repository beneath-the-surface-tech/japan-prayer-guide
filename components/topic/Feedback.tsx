import { Trans, useTranslation } from "next-i18next"
import React from "react"
import Button from "react-bootstrap/Button"

const Feedback = () => {
    const { t } = useTranslation("common")
    return (
        <section
            data-testid={"feedback-sect"}
            className="text-center m-0 p-4 w-100"
            style={{ backgroundColor: "#EAF4FF" }}
        >
            <h1 data-testid={"feedback-title"} className="text-primary fs-2">
                <Trans t={t} i18nKey="feedback.title" />
            </h1>
            <Button className="bg-secondary text-white border-0 p-2 w-100 mt-4" style={{ maxWidth: "300px" }}>
                <p data-testid={"feedback-button"} className="m-0">
                    <Trans t={t} i18nKey="feedback.button" />
                </p>
            </Button>
        </section>
    )
}

export default Feedback
