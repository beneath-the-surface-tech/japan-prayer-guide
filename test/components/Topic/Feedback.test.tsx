import { render, screen } from '@testing-library/react';
import Feedback from '../../../components/topic/Feedback';

describe("Feedback", () => {
    test("Renders a section with the right text inside", () => {
        const testText = "Hey how's it going"
        const testTitle = "Give us Feedback"
        render(<Feedback button={testText} title={testTitle} />)
        const feedbackSection = screen.getByTestId("feedback-sect")
        const feedbackButton = screen.getByTestId("feedback-button")
        const feedbackTitle = screen.getByTestId("feedback-title")

        expect(feedbackSection).toHaveClass("text-center", "m-0", "p-4")

        expect(feedbackButton).toHaveTextContent(testText)
        expect(feedbackButton).toHaveClass("m-0")

        expect(feedbackTitle).toHaveTextContent(testTitle)
        expect(feedbackTitle).toHaveClass("text-primary", "fs-2")
    })
})