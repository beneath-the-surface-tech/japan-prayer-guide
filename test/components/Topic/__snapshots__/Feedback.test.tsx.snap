// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-center m-0 p-4 w-100"
        data-testid="feedback-sect"
        style="background-color: rgb(234, 244, 255);"
      >
        <h1
          class="text-primary fs-2"
          data-testid="feedback-title"
        >
          feedback
        </h1>
        <button
          class="bg-secondary text-white border-0 p-2 w-100 mt-4 btn btn-primary"
          style="max-width: 300px;"
          type="button"
        >
          <p
            class="m-0"
            data-testid="feedback-button"
          >
            abc
          </p>
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-center m-0 p-4 w-100"
      data-testid="feedback-sect"
      style="background-color: rgb(234, 244, 255);"
    >
      <h1
        class="text-primary fs-2"
        data-testid="feedback-title"
      >
        feedback
      </h1>
      <button
        class="bg-secondary text-white border-0 p-2 w-100 mt-4 btn btn-primary"
        style="max-width: 300px;"
        type="button"
      >
        <p
          class="m-0"
          data-testid="feedback-button"
        >
          abc
        </p>
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
